src/include/catalog/index_schema.h

```diff
@@ -0,0 +1,198 @@
+     */
+    const std::string &Name() const { return name_; }
+
+    /**
+     * @return oid of this key column
+     */
+    indexkeycol_oid_t Oid() const { return oid_; }
+
+    /**
+     * @return definition expression
+     */
+    common::ManagedPointer<const parser::AbstractExpression> StoredExpression() const { return definition_; }
+
+    /**
+     * @warning only defined for varlen types
+     * @return maximum varlen size of this varlen column
+     */
+    uint16_t MaxVarlenSize() const { return static_cast<uint16_t>((packed_type_ & MASK_VARLEN) >> OFFSET_VARLEN); }
+
+    /**
+     * @return size of the attribute in bytes. Varlen attributes have the sign bit set.
+     */
+    uint8_t AttrSize() const { return type::TypeUtil::GetTypeSize(Type()); }
+
+    /**
+     * @return type of this key column
+     */
+    type::TypeId Type() const { return static_cast<type::TypeId>(packed_type_ & MASK_TYPE); }
+
+    /**
+     * @return true if this column is nullable
+     */
+    bool Nullable() const { return static_cast<bool>(packed_type_ & MASK_NULLABLE); }
+
+   private:
+    static constexpr uint32_t MASK_VARLEN = 0x00FFFF00;
+    static constexpr uint32_t MASK_NULLABLE = 0x00000080;
+    static constexpr uint32_t MASK_TYPE = 0x0000007F;
+    static constexpr uint32_t OFFSET_VARLEN = 8;
+
+    std::string name_;
+    indexkeycol_oid_t oid_;
+    uint32_t packed_type_;
+    common::ManagedPointer<const parser::AbstractExpression> definition_;
+    std::string serialized_expression_;
+
+    // TODO(John): Should these "OIDS" be implicitly set by the index in the columns?
+    void SetOid(indexkeycol_oid_t oid) { oid_ = oid; }
+
+    void SetMaxVarlenSize(uint16_t max_varlen_size) {
+      TERRIER_ASSERT((packed_type_ & MASK_VARLEN) == 0, "Should only set max varlen size once.");
+      const auto varlen_bits = (max_varlen_size << OFFSET_VARLEN) & MASK_VARLEN;
+      packed_type_ = packed_type_ | varlen_bits;
+    }
+
+    void SetTypeId(type::TypeId type_id) {
+      TERRIER_ASSERT((packed_type_ & MASK_TYPE) == 0, "Should only set type once.");
+      packed_type_ = packed_type_ | (static_cast<uint32_t>(type_id) & MASK_TYPE);
+    }
+
+    void SetNullable(bool nullable) {
+      TERRIER_ASSERT((packed_type_ & MASK_NULLABLE) == 0, "Should only set nullability once.");
+      packed_type_ = nullable ? packed_type_ | MASK_NULLABLE : packed_type_;
+    }
+
+    friend class DatabaseCatalog;
+    friend class postgres::Builder;
+
+    friend class tpcc::Schemas;
+    friend class terrier::StorageTestUtil;
+  };
+
+  /**
+   * Instantiates a new catalog description of an index
+   * @param columns describing the individual parts of the key
+   * @param is_unique indicating whether the same key can be (logically) visible repeats are allowed in the index
+   * @param is_primary indicating whether this will be the index for a primary key
+   * @param is_exclusion indicating whether this index is for exclusion constraints
```

Comment:
This exclusion parameter is missing from the list of constraints. Is it always false for now?

Reply:
Can you clarify what you mean by this?